Feat/stats design

shan-226 opened the pull request:
I redid the stats page's frontend so that it would look like the design on figma. You can access the page by clicking on the Mood box on the profile for now. This will change when the profile page has been redesigned. @pieryvdl and @u22520016 can you please review this when you get the chance.

u22520016 commented:
I really like it.  Just a few comments regarding the closeness with the Figma design:

1. The disclaimer on top needs padding and can be set to the grey text color we use for placeholders normally.  It could also be set to bold.
2. The heading of each stats section does not match the Figma design.  It needs to be bold.  Remove the fontStyle: 'Inter' line as that can cause problems with fontWeights and the default font is already 'Inter'.
3. There could be some more padding around the pie charts (like 12)
4. The 2 bottom green stat boxes at the top need to be the same width as the top 3 ones (padding 12 on both sides like on Figma)
5. The filter button only works when you click on the icon and not on the text as well.
6. The text inside the green boxes at the top should also be bolder

Good job overall though.

u22520016 commented on lib/view/components/stats/stats_total_time_boxes.dart:
```diff
@@ -0,0 +1,245 @@
+                        child: Text(
+                          'Today',
+                          style: TextStyle(
+                            fontFamily: 'Inter',
+                            fontSize: 15,
+                            fontWeight: FontWeight.bold,
+                            color: Theme.of(context).colorScheme.secondary,
```
The following should be:

`fontSize: 12,
fontWeight: FontWeight.bold
`

u22520016 reviewed (changes requested):
Please make those changes I requested.

shan-226 commented:
> I really like it. Just a few comments regarding the closeness with the Figma design:
> 
> 1. The disclaimer on top needs padding and can be set to the grey text color we use for placeholders normally.  It could also be set to bold.
> 2. The heading of each stats section does not match the Figma design.  It needs to be bold.  Remove the fontStyle: 'Inter' line as that can cause problems with fontWeights and the default font is already 'Inter'.
> 3. There could be some more padding around the pie charts (like 12)
> 4. The 2 bottom green stat boxes at the top need to be the same width as the top 3 ones (padding 12 on both sides like on Figma)
> 5. The filter button only works when you click on the icon and not on the text as well.
> 6. The text inside the green boxes at the top should also be bolder
> 
> Good job overall though.

1. It already had padding. I added more and changed the text color and made it bold.
2. The headers are already the same font size and weight as on figma. Removing the inter did nothing.
3. There was already 16, I increased it to 28.
4. The boxes all have left/right padding of 30. I can't really make them align unless I make them fixed lengths. I made them update their length according to the size of the screen so the top row of boxes won't always align, but will with the screen size of most standard phone sizes.
5. Fixed this.
6. They are already bold.

u22520016 reviewed (changes requested):
Okay @shan-226...I changed a few things myself as it is too hard to explain and it is quicker for me to do it myself.  Please have a look at my code.  I changed some subtle stuff but its mostly stuff I asked you to change.  One thing I noted is that you actually don't always follow the Figma design.  A lot of the times you gave some text the wrong font size or you go the padding completely wrong on some stuff.  Please follow the Figma design exactly, it's there for a reason.  If you decide you want to change something please ask us first, even if it is something small like padding.  One other thing that you should also strongly consider and should do is to separate the different parts of the page into different component files. I believe @MonikaTheiss already told you to do this.  The code becomes very unreadable and writable when you write all of the code in one file.  I also see that you use a lot of unnecessary styling stuff like changing the fonts.  I removed the font specifications and it did, in fact make all the text bold.  Please consider these points in the future.

pieryvdl reviewed (changes requested):
Hi Shannon, 

The images don't load properly on the emulator. 
![image](https://github.com/user-attachments/assets/f7d43757-9356-4774-809a-2b5822290c53)
It could be the way they are listed in the assets.

Is it possible to decrease the size of the ring? Of the Mood? I see it kind of squishes the text to the side and it makes the padding off. 
The genres "Filter" text and the "game type" eg. "Shooter" is also not bold like on the design. 

I unfortunately don't have any leaderboard stats to check it, but I trust that it should be fine. 

Thank you for your hard work Shannon, you are 95% there.

shan-226 commented:
I believe I fixed all the issues you brought up.

MonikaTheiss commented:
@shan-226 I'm struggling to find the stats... can you maybe guide me to where I can find them please?

shan-226 commented:
> @shan-226 I'm struggling to find the stats... can you maybe guide me to where I can find them please?

It is in the profile when you click on the mood container. I will update this for the new profile when it is in develop.

MonikaTheiss commented:
I believe the new profile is in develop and you pulled develop in before fixing the linter issues.. on my side I see the new profile and I cant find the mood button... could you maybe run it on your side to double check ( I'm sorry, I know its not ideal, but I'm confused) This is what I see on my side ono the stats design branch(sorry for the very big image):

![WhatsApp Image 2024-08-09 at 21 16 51_1f024447](https://github.com/user-attachments/assets/8b8df1bc-c8f6-4770-abd7-7af17248f8bf)

shan-226 commented:
> I believe the new profile is in develop and you pulled develop in before fixing the linter issues.. on my side I see the new profile and I cant find the mood button... could you maybe run it on your side to double check ( I'm sorry, I know its not ideal, but I'm confused) This is what I see on my side ono the stats design branch(sorry for the very big image):
> 
> ![WhatsApp Image 2024-08-09 at 21 16 51_1f024447](https://private-user-images.githubusercontent.com/131775579/356728055-8b8df1bc-c8f6-4770-abd7-7af17248f8bf.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMyMzE4NzAsIm5iZiI6MTcyMzIzMTU3MCwicGF0aCI6Ii8xMzE3NzU1NzkvMzU2NzI4MDU1LThiOGRmMWJjLWM4ZjYtNDc3MC1hYmQ3LTdhZjE3MjQ4ZjhiZi5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgwOVQxOTI2MTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lMThkYjU2MTBhNTlmZDM5ZDBhYTZjNzg5MzZkYzk0MTUzZDlmZDk2ZjJiMDU4NTZkMGNkNzcxMWVkODdlODBiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.BCLEnXchZHobfLTqzYuYFTtXAxkgYi6rA9be9s9yimE)

Oh my word. I didn't think the profile would have changed already since the profile is still a pull request and didn't check this. I will quickly fix this.

MonikaTheiss commented:
Thank you for adding the navigation ☺️ So a small thing on my side, and I see the design also has this. On the Mood pie chart, the yellow with white text does not look good, since its really difficult to see the number in white. I believe @u22520016 made the font color on the Happy (yellow) and disgusted (green)  pie slices  black so its easier to see. Could you perhaps change this please?

MonikaTheiss commented:
I'm also getting "error loading games" when clicking on the mood slices.. not too sure if that is supposed to happen?

MonikaTheiss commented:
> I'm also getting "error loading games" when clicking on the mood slices.. not too sure if that is supposed to happen?

every slice now returns game data except the sad slice... I remember this has been an ongoing issue. Have you looked into it yet?

pieryvdl commented:
Hi Shannon I also think the yellow and white does not go well. If you could change that too please. 😄 Then it seems my games data does load. When clicking on the yellow part of the pie chart.

shan-226 commented:
> Thank you for adding the navigation ☺️ So a small thing on my side, and I see the design also has this. On the Mood pie chart, the yellow with white text does not look good, since its really difficult to see the number in white. I believe @u22520016 made the font color on the Happy (yellow) and disgusted (green) pie slices black so its easier to see. Could you perhaps change this please?

Would it be possible for me to make all the numbers black instead of just 2 of them as I think its a bit weird if just 2 are a different color?

pieryvdl commented:
> Would it be possible for me to make all the numbers black instead of just 2 of them as I think its a bit weird if just 2 are a different color?

Yeah I think that would be best.

MonikaTheiss commented:
> > Thank you for adding the navigation ☺️ So a small thing on my side, and I see the design also has this. On the Mood pie chart, the yellow with white text does not look good, since its really difficult to see the number in white. I believe @u22520016 made the font color on the Happy (yellow) and disgusted (green) pie slices black so its easier to see. Could you perhaps change this please?
> 
> Would it be possible for me to make all the numbers black instead of just 2 of them as I think its a bit weird if just 2 are a different color? @MonikaTheiss , @pieryvdl , @u22520016

Please just double check with @u22520016 before changing anything, as that was the way his design was.

u22520016 commented:
> > Thank you for adding the navigation ☺️ So a small thing on my side, and I see the design also has this. On the Mood pie chart, the yellow with white text does not look good, since its really difficult to see the number in white. I believe @u22520016 made the font color on the Happy (yellow) and disgusted (green) pie slices black so its easier to see. Could you perhaps change this please?
> 
> Would it be possible for me to make all the numbers black instead of just 2 of them as I think its a bit weird if just 2 are a different color? @MonikaTheiss , @pieryvdl , @u22520016

It has to be different colors depending on the background color because it'll be hard to see white text on yellow background...that is why they are different colors.

shan-226 commented:
> > I'm also getting "error loading games" when clicking on the mood slices.. not too sure if that is supposed to happen?
> 
> every slice now returns game data except the sad slice... I remember this has been an ongoing issue. Have you looked into it yet?

I looked into fixing this. I spent 3 hours debugging it. I think I know what the problem is but I don't know how to fix it yet. I'll look into it today and tomorrow but for now, can I merge this branch in so that the page is in develop for the demo?

shan-226 commented:
I fixed all the things you requested. @MonikaTheiss, @pieryvdl and @u22520016 please review this when you get the chance.

MonikaTheiss commented:
> > > I'm also getting "error loading games" when clicking on the mood slices.. not too sure if that is supposed to happen?
> > 
> > 
> > every slice now returns game data except the sad slice... I remember this has been an ongoing issue. Have you looked into it yet?
> 
> I looked into fixing this. I spent 3 hours debugging it. I think I know what the problem is but I don't know how to fix it yet. I'll look into it today and tomorrow but for now, can I merge this branch in so that the page is in develop for the demo?

Please quickly join me and in a discord channel

MonikaTheiss commented:
all of the changes look good, thank you ! I have sent you a suggestion on how you can address the mood pie chart issue on Whatsapp.

MonikaTheiss commented:
Thank you @shan-226 , it works on my side 😄 🥳 I will allow @pieryvdl and @u22520016 to do the formal reviewing, but I'm happy , thank you

pieryvdl reviewed (approved):
The games work on my side. Thanks Shannon, you put in really hard work into this pull request. It is much appreciated!

u22520016 reviewed (approved):
t looks good thank you!
